### DISCUSSION TITLE
Fokker-Planck equation (SDE) with PyPDE?

### DISCUSSION THREAD

**Discussion by ivankukuljan** [zwicker-group/py-pde#383] [General]
Dear All,

I would like to use PyPDE to solve a 1D Fokker Planck equation:

![image](https://user-images.githubusercontent.com/108279515/226636163-a1ab4488-a032-4fb8-ae17-5ede4c4a341b.png)

for the general diffusion g and drift mu functions. The FP equation corresponds to the following stochastic process:
![image](https://user-images.githubusercontent.com/108279515/226636668-d60f0867-a44a-4b71-a241-f4e0f21297dc.png)

I would like to solve it with probability preserving boundary conditions:
![image](https://user-images.githubusercontent.com/108279515/226636321-c6c37199-ee58-4928-b894-8dce8c46748b.png)
where a and b are domain boundaries.

I've expanded the terms in the FP equation and tried implementing it in the package using the code below (for the moment using simple Dirichlet boundary conditions) however that doesn't seem to work. What am I doing wrong? What would be the best way to implement the FP equation? Also, from your documentation it's not really clear how dot and * are defined (are they a scalar product and an element-wise product?) and when to use each type of multiplication.

```
def FokkerPlanckSolvePde(mu, g, bc, initial_cond, grid, t_max, dt):
   
    g_square = f" ({g})**2 "
   
    term_1= f" ( 0.5 * laplace({g_square}) - gradient({mu}) ) * c "
    term_2= f" dot( ( gradient({g_square}) - ({mu}) ) , gradient(c) )"
    term_3= f" 0.5 * ({g_square})  * laplace(c) "
   
    c_string = f" {term_1} + {term_2} + {term_3} "

    eq = PDE({"c": c_string}, bc=bc)

    result = eq.solve(initial_cond, t_max, dt=dt)  # solve the PDE
    return result
```

Kind regards,
Ivan

**Comment by david-zwicker** [comment]
I'm not quite sure what error you're seeing. What do you get as a result and what do you expect? Moreover, it would be helpful to have a minimal, self-consistent example that we could use to recreate the problem.
